lib/chat_widget.dart

```diff
@@ -19,6 +22,35 @@ class SingleChat extends StatefulWidget {
 
 class _SingleChatState extends State<SingleChat> {
   bool actionsExpanded = false;
+  bool imageSelected = false;
```

Comment:
if image is nullable then this could be `bool get _isImageSelected => image != null` which would prevent any chance the two connected values could get out of sync. Also bools are named in the form of a question like that.